openedx/core/djangoapps/external_user_ids/models.py

```diff
@@ -36,3 +46,52 @@ class ExternalId(TimeStampedModel):
     external_id_type = models.ForeignKey(ExternalIdType, db_index=True, on_delete=models.CASCADE)
     user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
     history = HistoricalRecords()
+
+    class Meta(object):
+        app_label = "external_id"
+
+    @classmethod
+    def user_has_external_id(cls, user, type_name):
+        """
+        Checks if a user has an ExternalId of the type_name provided
+        Arguments:
+            user: User to search for
+            type_name (str): Name of the type of ExternalId
+        """
+        try:
+            type_obj = ExternalIdType.objects.get(name=type_name)
+        except ExternalIdType.DoesNotExist:
+            LOGGER.info('No external id type of {}'.format(type_name))
+            return False
+        try:
+            external_id = cls.objects.get(
+                user=user,
+                external_id_type=type_obj
+            )
+            if external_id:
+                return True
+        except ExternalId.DoesNotExist:
+            return False
+
+    @classmethod
+    def add_new_user_id(cls, user, type_name):
+        """
+        Creates an ExternalId for the User of the type_name provided
+        Arguments:
+            user: User to create the ID for
+            type_name (str): Name of the type of ExternalId
+        """
+        try:
+            type_obj = ExternalIdType.objects.get(name=type_name)
+        except ExternalIdType.DoesNotExist:
+            LOGGER.info(
+                'External Creation failed for user {user}, no external id type of {type}'.format(
+                    user=user.id,
+                    type=type_name
+                )
+            )
+            return None
+        return cls.objects.create(
```

Comment:
What if a user:type pair already exists? I imagine we only want one of each otherwise the data gets screwy, so we'll probably want to check for that and add a `unique_together` to the model on those fields.

Reply:
I was thinking about adding the unique_together as well. That should be enough though right?

Reply:
I could also just use a `get_or_create`